raster-foundry/raster-foundry
# Try to fetch pre-generated tifs for low zoom level tile requests

## Overview

This PR goes looking for a pre-generated overview image if the zoom in a mosaic TMS request is
below a certain threshold. If it's not below that threshold or something goes wrong in that process,
it falls back to the current strategy for fetching and rendering the image.

### Checklist


## Notes

The path in s3 might not be 100% correct -- there's simultaneous work going on there -- but I'll update that after #4936 is merged.

This also doesn't try to fetch overviews for tms requests in the analysis context. That doesn't seem to come up as much for some reason because they're not in the preview windows, however similar logic could be added if we wanted to.  

## Testing Instructions

- look at some tiles at zoom levels >= 6
- they should look normal
- Try to look at any project at zoom level 5 or lower
- It should succeed and warn you that it failed to find an overview
- Mess with the environment variable for the threshold for searching for overviews and restart your server
- Behavior should change according to your wishes

Closes #4838
